### PR TITLE
Add $psversiontable output to Diagnostic Logs

### PR DESCRIPTION
**Pull request by stephenmichaelf**

### CODE REVIEW

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 1]
```diff
@@ -209,9 +213,7 @@ private static bool IsDefenderEnabled()
         {
             return Process.GetProcessesByName("MsMpEng.exe").FirstOrDefault() != null;
         }
-#endif
 
-#if OS_WINDOWS
         // Returns whether or not the Windows firewall is enabled.
         private static bool IsFirewallEnabled()
```
Comment (TingluoHuang) [inline src/Agent.Worker/DiagnosticLogManager.cs]: why `IsFirewallEnabled()` is `static`  :D

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 2]
```diff
@@ -234,6 +236,39 @@ private static bool IsFirewallEnabled()
...
+
+        private async Task<string> GetPsVersionInfo()
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = @"Write-Host ($psversiontable | Out-String)";
+            using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+            {
+                processInvoker.OutputDataReceived += (object sender, ProcessDataReceivedEventArgs args) =>
```
Comment (TingluoHuang) [inline src/Agent.Worker/DiagnosticLogManager.cs]: i would just print both STDOUT/STDERR to output and not to `Trace`

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 3]
```diff
@@ -234,6 +236,39 @@ private static bool IsFirewallEnabled()
                 return false;
             }
         }
+
+        private async Task<string> GetPsVersionInfo()
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = @"Write-Host ($psversiontable | Out-String)";
```
Comment (TingluoHuang) [inline src/Agent.Worker/DiagnosticLogManager.cs]: I prefer `$PSVersionTable`  :)

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 4]
```diff
@@ -234,6 +236,39 @@ private static bool IsFirewallEnabled()
...
+                };
+                
+                await processInvoker.ExecuteAsync(
+                    workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Bin),
+                    fileName: powerShellExe,
+                    arguments: arguments,
+                    environment: null,
+                    requireExitCodeZero: false,
+                    outputEncoding: null,
+                    killProcessOnCancel: true,
```
Comment (TingluoHuang) [inline src/Agent.Worker/DiagnosticLogManager.cs]: killProcessOnCancel: false,

Comment (stephenmichaelf) [reply src/Agent.Worker/DiagnosticLogManager.cs]: Fixed.

File: src/Agent.Worker/DiagnosticLogManager.cs (hunk 3 above)
Comment (stephenmichaelf) [reply src/Agent.Worker/DiagnosticLogManager.cs]: Fixed.

File: src/Agent.Worker/DiagnosticLogManager.cs (hunk 1 above)
Comment (stephenmichaelf) [reply src/Agent.Worker/DiagnosticLogManager.cs]: Fixed. :) Good catch.

File: src/Agent.Worker/DiagnosticLogManager.cs (hunk 2 above)
Comment (stephenmichaelf) [reply src/Agent.Worker/DiagnosticLogManager.cs]: executionContext.Output?

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 5]
```diff
@@ -234,6 +236,39 @@ private static bool IsFirewallEnabled()
...
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = @"Write-Host ($PSVersionTable | Out-String)";
+            using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+            {
+                processInvoker.OutputDataReceived += (object sender, ProcessDataReceivedEventArgs args) =>
+                {
+                    executionContext.Output($"STDOUT: {args.Data}");
```
Comment (TingluoHuang) [inline src/Agent.Worker/DiagnosticLogManager.cs]: just builder.AppendLine(args.Data); for both STDOUT/STDERR.
we don't need this to executionContext.Output().

Comment (stephenmichaelf) [reply src/Agent.Worker/DiagnosticLogManager.cs]: Done. Ready for review.

**Comment by stephenmichaelf** [comment]
Removed unnecessary variable.

**Pull request merged**